tests/foreman/ui/test_discoveredhost.py

```diff
@@ -854,22 +874,26 @@ def test_positive_provision_without_auto_provision(self):
                 host_limit=1,
                 hostgroup=self.config_env['host_group'],
                 search_rule='cpu_count = 1',
-                locations=[self.loc_name],
+                locations=[self.loc.name],
             )
             self.assertIsNotNone(self.discoveryrules.search(rule_name))
             with LibvirtGuest() as pxe_host:
                 host_name = pxe_host.guest_name
-                self.discoveredhosts.assertdiscoveredhost(host_name)
+                self.assertTrue(
+                    self.discoveredhosts.waitfordiscoveredhost(host_name)
+                )
                 self.assertIsNotNone(self.discoveredhosts.search(host_name))
                 # Check that host shouldn't list under all hosts
-                search = self.hosts.search(
-                    u'{0}.{1}'.format(host_name, self.config_env['domain'])
-                )
-                self.assertIsNone(search)
+                self.assertIsNone(self.hosts.search(
+                    u'{0}.{1}'.format(host_name, self.config_env['domain'])))
                 # Check that host still listed under discovered hosts
                 self.assertIsNotNone(self.discoveredhosts.search(host_name))
+                # Revert the discovery_auto flag to default value
+                discovery_auto.value = default_discovery_auto
+                discovery_auto.update(['value'])
```

Comment:
This revert won't work if the test will fail on some assertion above. Please use `try: ... finally:` or cleanup func here